src/modules/lua/script_lua.c

```diff
@@ -782,30 +744,157 @@ static void luaReplyToServerReply(client *c, client *script_client, lua_State *l
                 lua_pop(lua, 1);
                 break;
             }
-            luaReplyToServerReply(c, script_client, lua);
+            luaReplyToServerReply(ctx, resp_version, lua);
             mbulklen++;
         }
-        setDeferredArrayLen(c, replylen, mbulklen);
+        ValkeyModule_ReplySetArrayLength(ctx, mbulklen);
         break;
-    default: addReplyNull(c);
+    default: ValkeyModule_ReplyWithNull(ctx);
     }
     lua_pop(lua, 1);
 }
 
 /* ---------------------------------------------------------------------------
  * Lua server.* functions implementations.
  * ------------------------------------------------------------------------- */
-void freeLuaServerArgv(robj **argv, int argc, int argv_len);
+void freeLuaServerArgv(ValkeyModuleCtx *ctx, ValkeyModuleString **argv, int argc);
+
+/* Return the number of digits of 'v' when converted to string in radix 10.
+ * See ll2string() for more information. */
+static uint32_t digits10(uint64_t v) {
+    if (v < 10) return 1;
+    if (v < 100) return 2;
+    if (v < 1000) return 3;
+    if (v < 1000000000000UL) {
+        if (v < 100000000UL) {
+            if (v < 1000000) {
+                if (v < 10000) return 4;
+                return 5 + (v >= 100000);
+            }
+            return 7 + (v >= 10000000UL);
+        }
+        if (v < 10000000000UL) {
+            return 9 + (v >= 1000000000UL);
+        }
+        return 11 + (v >= 100000000000UL);
+    }
+    return 12 + digits10(v / 1000000000000UL);
+}
+
+/* Convert a unsigned long long into a string. Returns the number of
+ * characters needed to represent the number.
+ * If the buffer is not big enough to store the string, 0 is returned.
+ *
+ * Based on the following article (that apparently does not provide a
+ * novel approach but only publicizes an already used technique):
+ *
+ * https://www.facebook.com/notes/facebook-engineering/three-optimization-tips-for-c/10151361643253920 */
```

Comment:
This link doesn't seem to work anymore (or it requires login). We can skip it or use wayback machine. I found a link that works:

```suggestion
 * https://web.archive.org/web/20150427221229/https://www.facebook.com/notes/facebook-engineering/three-optimization-tips-for-c/10151361643253920 */
```

We could also just delete the link.

Reply:
I'll keep the archive link